bin/parse-legacy-file

```diff
@@ -3,18 +3,26 @@
 set -Eeuo pipefail
 
 get_legacy_version() {
-  local -r current_plugin="$1"
-  local -r version_file="$2"
-  local -r plugins_dir="${ASDF_DATA_DIR:-$HOME/.asdf}/plugins"
+  local -r version_file="$1"
   local -r supported_plugins=(terraform packer)
 
   # check if current plugin is supported and with a readable legacy file
   for plugin in "${supported_plugins[@]}"; do
-    if [[ ${current_plugin} =~ ${plugins_dir}/${plugin} &&
-      ${version_file} == *".${plugin}-version" && -r ${version_file} ]]; then
+    if [[ ${version_file} == *".${plugin}-version" && -r ${version_file} ]]; then
       cat "${version_file}"
     fi
   done
+
+  if [[ "${version_file}" =~ "main.tf"  ]]; then
+    grep --no-filename \
+         --recursive \
+         --fixed-strings \
+         --exclude-dir .terraform \
+         required_version \
+         "$(dirname "${version_file}")" \
+      | sed -e 's/required_version\s*=\s*"=\(.*\)"/\1/' \
+      | tr -d ' '
```

Comment:
Same sort of scenario as the other comment above, this breaks when one of the other legacy files is present in the same dir as a `main.tf` file.